DSharpPlus.UnifiedCommands/Application/Internals/ApplicationHandler.cs

```diff
@@ -0,0 +1,143 @@
+using System.Reflection;
+using DSharpPlus.Entities;
+using DSharpPlus.UnifiedCommands.Application.Conditions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace DSharpPlus.UnifiedCommands.Application.Internals;
+
+internal class ApplicationHandler
+{
+    private DiscordClient _client;
+    private ApplicationModule _module = null!;
+    private DiscordInteraction _interaction;
+    private ApplicationMethodData _data;
+    private readonly IReadOnlyList<Func<IServiceProvider, IApplicationCondition>> _conditionBuilders;
+    private object?[]? _args;
+    private IServiceScope _scope;
+
+    internal ApplicationHandler(ApplicationMethodData data, DiscordInteraction interaction,
+        IReadOnlyList<Func<IServiceProvider, IApplicationCondition>> conditionBuilders, object?[]? args,
+        IServiceScope scope, DiscordClient client)
+    {
+        _data = data;
+        _interaction = interaction;
+        _conditionBuilders = conditionBuilders;
+        _args = args;
+        _scope = scope;
+        _client = client;
+    }
+
+    internal async Task TurnResultIntoActionAsync(IApplicationResult result)
+    {
+        switch (result.Type)
+        {
+            case ApplicationResultType.Reply:
+                DiscordInteractionResponseBuilder builder = new();
+                if (result.Content is not null)
+                {
+                    builder.WithContent(result.Content);
+                }
+
+                if (result.Embeds is not null)
+                {
+                    builder.AddEmbeds(result.Embeds);
+                }
+
+                await _module.Interaction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource,
+                    builder);
+                break;
+            case ApplicationResultType.FollowUp:
+                DiscordFollowupMessageBuilder followUpBuilder = new();
+                if (result.Content is not null)
+                {
+                    followUpBuilder.WithContent(result.Content);
+                }
+
+                if (result.Embeds is not null)
+                {
+                    followUpBuilder.AddEmbeds(result.Embeds);
+                }
+
+                await _module.Interaction.CreateFollowupMessageAsync(followUpBuilder);
+                break;
+        }
+    }
+
+    internal async Task BuildModuleAndExecuteCommandAsync()
+    {
+        try
+        {
+            _module = (ApplicationModule)_data.Module.Factory.Invoke(_scope.ServiceProvider, null);
+            _module.Interaction = _interaction;
+            _module.Client = _client;
+            _module._handler = this;
+
+            foreach (Func<IServiceProvider, IApplicationCondition> conditionBuilder in _conditionBuilders)
```

Comment:
1 <sub>Not to be grammar police, but I couldn't find the comment because</sup> Synchronous
2 Sequential, not synchronous, though the choice to do a task and then await it is...interesting. cc @akiraveliara

3 I think it's fine; most common case is there's zero or one conditions

Reply:
Reason for this design was cause conditions might depend on scope changes that other conditions may do. So adding every task to a list and await them all might be a bad decision.

Reply:
Fair enough